backend/packages/Upgrade/src/api/services/FeatureFlagService.ts

```diff
@@ -110,36 +140,51 @@ export class FeatureFlagService {
         throw error;
       }
 
-      // creating variations docs
-      const variationDocsToSave =
-        variations &&
-        variations.length > 0 &&
-        variations.map((variation: FlagVariation) => {
-          variation.id = variation.id || uuid();
-          variation.featureFlag = featureFlagDoc;
-          return variation;
-        });
-
-      // saving variations
-      let variationDocs: FlagVariation[];
+      const {
+        segmentExists: includeSegmentExists,
+        segmentDoc: segmentIncludeDoc,
+        segmentDocToSave: segmentIncludeDocToSave,
+      } = await this.addPrivateSegmentToDB(featureFlagSegmentInclusion, flag, 'Inclusion', logger);
+      const {
+        segmentExists: excludeSegmentExists,
+        segmentDoc: segmentExcludeDoc,
+        segmentDocToSave: segmentExcludeDocToSave,
+      } = await this.addPrivateSegmentToDB(featureFlagSegmentExclusion, flag, 'Exclusion', logger);
+
+      let featureFlagSegmentInclusionDoc: FeatureFlagSegmentInclusion;
+      let featureFlagSegmentExclusionDoc: FeatureFlagSegmentExclusion;
+
       try {
-        variationDocs = await this.flagVariationRepository.insertVariations(
-          variationDocsToSave,
-          transactionalEntityManager
-        );
+        [featureFlagSegmentInclusionDoc, featureFlagSegmentExclusionDoc] = await Promise.all([
+          includeSegmentExists
+            ? this.featureFlagSegmentInclusionRepository.insertData(
+                segmentIncludeDocToSave,
+                logger,
+                transactionalEntityManager
+              )
+            : (Promise.resolve([]) as any),
+          excludeSegmentExists
+            ? this.featureFlagSegmentExclusionRepository.insertData(
+                segmentExcludeDocToSave,
+                logger,
+                transactionalEntityManager
+              )
+            : (Promise.resolve([]) as any),
+        ]);
       } catch (err) {
-        const error = new Error(`Error in creating variation "addFeatureFlagInDB" ${err}`);
-        (error as any).type = SERVER_ERROR.QUERY_FAILED;
+        const error = err as Error;
+        error.message = `Error in creating inclusion or exclusion segments "addFeatureFlagInDB"`;
         logger.error(error);
         throw error;
       }
 
-      const variationDocToReturn = variationDocs.map((variationDoc) => {
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const { featureFlagId, ...rest } = variationDoc as any;
-        return rest;
-      });
-      return { ...featureFlagDoc, variations: variationDocToReturn as any };
+      const newFeatureFlagObject = {
+        ...featureFlagDoc,
+        featureFlagSegmentInclusion: { ...featureFlagSegmentInclusionDoc, segment: segmentIncludeDoc } as any,
```

Comment:
This will depend on if `includeSegmentExists` and the same for `featureFlagSegmentExclusion`

Reply:
True. Good catch.